<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1354</width>
    <height>579</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGLDemo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="MyGL" name="mygl">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>11</y>
      <width>711</width>
      <height>521</height>
     </rect>
    </property>
   </widget>
   <widget class="QTabWidget" name="ModeTabWidget">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>10</y>
      <width>611</width>
      <height>521</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="MeshModeTab">
     <attribute name="title">
      <string>Mesh Mode</string>
     </attribute>
     <widget class="QListWidget" name="vertsListWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>111</width>
        <height>261</height>
       </rect>
      </property>
     </widget>
     <widget class="QListWidget" name="halfEdgesListWidget">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>10</y>
        <width>111</width>
        <height>261</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>370</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Face Color</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="vertPosYSpinBox">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>320</y>
        <width>62</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <double>-99.989999999999995</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>390</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Green</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QListWidget" name="facesListWidget">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>10</y>
        <width>111</width>
        <height>261</height>
       </rect>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="faceBlueSpinBox">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>370</y>
        <width>62</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.050000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>280</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Faces</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="vertPosXSpinBox">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>320</y>
        <width>62</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <double>-99.989999999999995</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>390</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Red</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>340</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Y</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>340</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>X</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>340</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Z</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>280</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Half-Edges</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="BN_MarkSharp">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>310</y>
        <width>81</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Mark Sharp</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="faceGreenSpinBox">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>370</y>
        <width>62</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.050000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>320</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Vertex Position</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="vertPosZSpinBox">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>320</y>
        <width>62</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <double>-99.989999999999995</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>280</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Vertices</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="faceRedSpinBox">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>370</y>
        <width>62</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.050000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>390</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Blue</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QTabWidget" name="ModifierTabWidget">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>10</y>
        <width>191</width>
        <height>261</height>
       </rect>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Edge</string>
       </attribute>
       <widget class="QWidget" name="verticalLayoutWidget">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>20</y>
          <width>181</width>
          <height>56</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QPushButton" name="BN_AddVertex">
           <property name="text">
            <string>Add Vertex</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="BN_InsertEdgeLoop">
           <property name="text">
            <string>Insert EdgeLoop</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Face</string>
       </attribute>
       <widget class="QWidget" name="verticalLayoutWidget_2">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>20</y>
          <width>181</width>
          <height>86</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QPushButton" name="BN_TriangulateFace">
           <property name="text">
            <string>Triangulate Face</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="BN_ExtrudeFace">
           <property name="text">
            <string>Extrude Face</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="BN_Bevel">
           <property name="text">
            <string>Bevel</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Mesh</string>
       </attribute>
       <widget class="QWidget" name="verticalLayoutWidget_3">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>20</y>
          <width>181</width>
          <height>31</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QPushButton" name="BN_Subdivision">
           <property name="text">
            <string>Subdivision</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
     <widget class="QPushButton" name="BN_Skinning">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>440</y>
        <width>88</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Skinning</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="SkeletonModeTab">
     <attribute name="title">
      <string>Skeleton Mode</string>
     </attribute>
     <widget class="QTreeWidget" name="SkeletonTree">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>256</width>
        <height>281</height>
       </rect>
      </property>
      <column>
       <property name="text">
        <string notr="true">Skeleton Tree</string>
       </property>
      </column>
     </widget>
     <widget class="QPushButton" name="BN_UnBind">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>440</y>
        <width>88</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>UnBind</string>
      </property>
     </widget>
     <widget class="QTabWidget" name="SkeletonEditTab">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>10</y>
        <width>321</width>
        <height>281</height>
       </rect>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Edit Joints</string>
       </attribute>
       <widget class="QWidget" name="gridLayoutWidget_2">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>10</y>
          <width>271</width>
          <height>94</height>
         </rect>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="3">
          <widget class="QDoubleSpinBox" name="jointRotZSpinBox">
           <property name="minimum">
            <double>-5.000000000000000</double>
           </property>
           <property name="maximum">
            <double>5.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.500000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QDoubleSpinBox" name="jointPosZSpinBox">
           <property name="minimum">
            <double>-5.000000000000000</double>
           </property>
           <property name="maximum">
            <double>5.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.500000000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="jointRotXSpinBox">
           <property name="minimum">
            <double>-5.000000000000000</double>
           </property>
           <property name="maximum">
            <double>5.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.500000000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_16">
           <property name="text">
            <string>Rotation:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QLabel" name="label_20">
           <property name="text">
            <string>Z</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QDoubleSpinBox" name="jointRotYSpinBox">
           <property name="minimum">
            <double>-5.000000000000000</double>
           </property>
           <property name="maximum">
            <double>5.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.500000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QDoubleSpinBox" name="jointPosYSpinBox">
           <property name="minimum">
            <double>-5.000000000000000</double>
           </property>
           <property name="maximum">
            <double>5.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.500000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QLabel" name="label_22">
           <property name="text">
            <string>Y</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>Location:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="label_19">
           <property name="text">
            <string>Y</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="3" column="3">
          <widget class="QLabel" name="label_23">
           <property name="text">
            <string>Z</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="jointPosXSpinBox">
           <property name="minimum">
            <double>-5.000000000000000</double>
           </property>
           <property name="maximum">
            <double>5.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.500000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="label_18">
           <property name="text">
            <string>X</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLabel" name="label_21">
           <property name="text">
            <string>X</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>Edit Weight</string>
       </attribute>
       <widget class="QWidget" name="horizontalLayoutWidget">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>50</y>
          <width>261</width>
          <height>132</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>Joint ID</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="Joint0_Name">
             <item>
              <property name="text">
               <string>None</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="Joint1_Name">
             <item>
              <property name="text">
               <string>None</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="Joint2_Name">
             <item>
              <property name="text">
               <string>None</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="Joint3_Name">
             <item>
              <property name="text">
               <string>None</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>Joint Weight</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="Joint0_weight">
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.050000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="Joint1_weight">
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.050000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="Joint2_weight">
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.050000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="Joint3_weight">
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.050000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1354</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_OBJ"/>
    <addaction name="actionLoad_Skeleton"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionCamera_Controls"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionCamera_Controls">
   <property name="text">
    <string>Camera Controls</string>
   </property>
  </action>
  <action name="actionLoad_OBJ">
   <property name="text">
    <string>Load OBJ</string>
   </property>
  </action>
  <action name="actionLoad_Skeleton">
   <property name="text">
    <string>Load Skeleton</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGL</class>
   <extends>QOpenGLWidget</extends>
   <header>mygl.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
